analyzers/src/SonarAnalyzer.CSharp/Rules/CastShouldNotBeDuplicated.cs

```diff
@@ -26,55 +27,205 @@
+            if (!(isPatternExpression.SyntaxNode.GetFirstNonParenthesizedParent() is IfStatementSyntax parentIfStatement))
             {
                 return;
             }
+            ProcessPatternExpression(analysisContext,
+                                     isPatternExpression.Pattern,
+                                     isPatternExpression.Expression,
+                                     isPatternExpression.SyntaxNode.GetLocation(),
+                                     parentIfStatement.Statement,
+                                     ReplaceWithAsAndNullCheckMessage);
+        }
 
-            var isExpressionLeftSymbol = analysisContext.SemanticModel.GetSymbolInfo(isExpression.Left).Symbol;
-            if (isExpressionLeftSymbol == null)
+        private static void IsExpression(SyntaxNodeAnalysisContext analysisContext)
+        {
+            var isExpression = (BinaryExpressionSyntax)analysisContext.Node;
+
+            if (!(isExpression.Right is TypeSyntax castType)
+                || !(isExpression.GetFirstNonParenthesizedParent() is IfStatementSyntax parentIfStatement)
+                || !(analysisContext.SemanticModel.GetSymbolInfo(castType).Symbol is INamedTypeSymbol castTypeSymbol)
+                || castTypeSymbol.TypeKind == TypeKind.Struct)
             {
                 return;
             }
 
-            var duplicatedCastLocations = parentIfStatement.Statement
+            ReportPatternAtMainVariable(analysisContext, isExpression.Left, isExpression.GetLocation(), parentIfStatement.Statement, castType, ReplaceWithAsAndNullCheckMessage);
+        }
+
+        private static List<Location> GetDuplicatedCastLocations(SyntaxNodeAnalysisContext analysisContext, SyntaxNode parentStatement, TypeSyntax castType, SyntaxNode typedVariable)
+        {
+            var typeExpressionSymbol = analysisContext.SemanticModel.GetSymbolInfo(typedVariable).Symbol
+                                       ?? analysisContext.SemanticModel.GetDeclaredSymbol(typedVariable);
+            if (typeExpressionSymbol == null)
+            {
+                return new List<Location>();
+            }
+
+            return parentStatement
                 .DescendantNodes()
                 .OfType<CastExpressionSyntax>()
-                .Where(x => x.Type.IsEquivalentTo(castType) && IsCastOnSameSymbol(x))
-                .Select(x => x.GetLocation());
+                .Where(x => x.Type.WithoutTrivia().IsEquivalentTo(castType.WithoutTrivia())
+                            && IsCastOnSameSymbol(x))
+                .Select(x => x.GetLocation()).ToList();
+
+            bool IsCastOnSameSymbol(CastExpressionSyntax castExpression) =>
+                Equals(analysisContext.SemanticModel.GetSymbolInfo(castExpression.Expression).Symbol, typeExpressionSymbol);
+        }
+
+        private static void ProcessPatternExpression(SyntaxNodeAnalysisContext analysisContext,
+            SyntaxNode isPattern,
```

Comment:
For some methods you aligned the parameters under each other and on some places not. See line 156. As we still do not have a clear coding style for that I would only ask you to align them to be consistent. For whichever option you opt.

Reply:
Done!